src/Forms/SearchableDropdownTrait.php

```
@@ -0,0 +1,594 @@
+     */
+    public function getValueArray(): array
+    {
+        $value = $this->Value();
+        if (empty($value)) {
+            return [];
+        }
+        if (is_array($value)) {
+            $arr = $value;
+            // Normalise FormBuilder values to be like Page EditForm values
+            //
+            // Page EditForm $values for non-multi field will be
+            // [
+            //   0 => '10',
+            // ];
+            // FormBuilder $values for non-multi field will be
+            // [
+            //   'label' => 'MyTitle15', 'value' => '10'
+            // ]
+            if (array_key_exists('value', $arr)) {
+                $arr = [$arr['value']];
+            }
+            // Page EditForm $values for multi will be
+            // [
+            //   0 => '10',
+            //   1 => '15'
+            // ];
+            // FormBuilder $values for multi will be
+            // [
+            //   0 => ['label' => '10', 'value' => 'MyTitle10', 'selected' => false],
+            //   1 => ['label' => '15', 'value' => 'MyTitle15', 'selected' => false]
+            // ];
+            if (array_key_exists(0, $arr) && is_array($arr[0]) && array_key_exists('value', $arr[0])) {
+                $arr = array_map(fn ($a) => $a['value'], $arr);
+            }
+            $arr = array_filter($arr);
+            return array_map(fn ($a) => (int) $a, $arr);
+        }
+        try {
+            // things stdClass cannot be cast to string
+            (string) $value;
+        } catch (Error $e) {
+            $value = 0;
+        }
+        if (ctype_digit((string) $value) && $value != 0) {
+            return [(int) $value];
+        }
+        if ($value instanceof SS_List) {
+            return array_filter($value->column('ID'));
+        }
+        if ($value instanceof DataObject && $value->exists()) {
+            return [$value->ID];
+        }
+        // Don't know what value is, handle gracefully
+        return [];
```

Discarding the value doesn't seem like a graceful way to handle it - we should probably throw an exception instead?

---

I considered that, though because this is a form field used in the CMS it could leave CMS editors in a situation where the CMS just breaks and there's nothing that can do about it.

---

That makes sense.
In that case we should use `user_error($message, E_USER_WARNING)`, or get the logger via `Injector::inst()->get(LoggerInterface::class)` and log a warning.

---

Done